getting_started/scripting/gdscript/gdscript_styleguide.rst

```diff
@@ -688,36 +688,42 @@ Static typing
 
 Since Godot 3.1, GDScript supports :ref:`optional static typing<doc_gdscript_static_typing>`.
 
-Type hints
-~~~~~~~~~~
+Declared Types
+~~~~~~~~~~~~~~
 
-Place the colon right after the variable's name, without a space, and let the
-GDScript compiler infer the variable's type when possible.
+To declare a variable's type, use ``<variable>: <type>``:
 
-**Good**:
+::
+
+   var health: int = 0
+
+To declare the return type of a function, use ``-> <type>``: 
 
 ::
 
-   onready var health_bar: ProgressBar = get_node("UI/LifeBar")
+   func heal(amount: int) -> void:
 
-   var health := 0 # The compiler will use the int type.
+Inferred Types
+~~~~~~~~~~~~~~
 
-**Bad**:
+In some cases you can let the compiler infer the type, using ``:=``:
 
 ::
+   var health := 0  # The compiler will use the int type.
 
-   # The compiler can't infer the exact type and will use Node
-   # instead of ProgressBar.
-   onready var health_bar := get_node("UI/LifeBar")
+However, in other cases the compiler is not able to infer the exact type, 
```

Comment:
It's not in many cases. It's only when the context is missing, e.g. you're in a GDScript file, you store an arbitrary node in a variable, and the target node isn't loaded in memory. In that case, the compiler falls back on the function's return type. For `get_node`, it's `Node`, but it could be something else if you're using your own wrapper function somewhere.

Reply:
Thanks for the feedback! clarified the `get_node` case: 
512bd80